packages/@react-aria/calendar/src/useCalendarBase.ts

```diff
@@ -39,36 +39,42 @@ export function useCalendarBase(props: CalendarPropsBase & DOMProps, state: Cale
   // Kinda hacky, but it's the easiest way to get the id from the cell.
   calendarIds.set(state, calendarId);
 
+  let focusFocusedDateCell = useCallback(() => {
+    const focusedCell = calendarBody.current.querySelector(`#${getCellId(state.focusedDate, calendarId)}`);
+    if (focusedCell && focusedCell !== document.activeElement) {
+      focusedCell.focus();
+    }
+  }, [state.focusedDate, calendarId]);
+
   useEffect(() => {
     // focus the calendar body when mounting
     if (autoFocus) {
-      calendarBody.current.focus();
+      focusFocusedDateCell();
     }
   }, [autoFocus]);
 
   // Announce when the current month changes
   useUpdateEffect(() => {
-    announce(monthFormatter.format(state.currentMonth));
+    if (state.isFocused) {
+      focusFocusedDateCell();
+    } else {
+      announce(monthFormatter.format(state.currentMonth));
+    }
   }, [state.currentMonth]);
 
   // Announce when the selected value changes
   useUpdateEffect(() => {
     if (selectedDateDescription) {
-      announce(selectedDateDescription);
+      announce(selectedDateDescription, 'polite', 3000);
     }
   }, [selectedDateDescription]);
 
-  // Ensure that the focused date is announced when it changes.
-  // VoiceOver does not announce changes to aria-activedescendant or move the VoiceOver cursor,
-  // so we blur and re-focus the calendar body to force it to do so.
-  // This will cause the focused date, along with the selected date (part of the table caption)
-  // to be announced as you move around the calendar.
+  // Ensure that the focused date moves focus to the focused date cell within the calendar.
   useUpdateEffect(() => {
-    if (document.activeElement === calendarBody.current) {
-      calendarBody.current.blur();
-      calendarBody.current.focus();
+    if (state.isFocused) {
+      focusFocusedDateCell();
     }
-  }, [state.focusedDate]);
+  }, [state.focusedDate, state.isFocused, focusFocusedDateCell]);
```

Comment:
this useUpdateEffect hurts my head, why/how is it different than the one on line 58 other than a different set of dependencies and an announce. I would like a plain english for why it's needed in addition to the other one

Reply:
The one at line 58 is dealing with the change in `state.currentMonth`, like when you press the previous or next month button. This one handles the date change, announcing the newly focused date in full when it receives focus.

Reply:
Cool, lets add a comment on each of these saying just that